fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/HashJoinCostModel.java

```diff
@@ -87,7 +87,8 @@ public double getCpuCost() {
         double probeCost;
         double leftOutput = leftStatistics.getOutputSize(context.getChildOutputColumns(0));
         double rightOutput = rightStatistics.getOutputSize(context.getChildOutputColumns(1));
-        int parallelFactor = Math.max(ConnectContext.get().getAliveBackendNumber(),
+        int parallelFactor = Math.max(ConnectContext.get().getAliveBackendNumber() +
```

Comment:
Thanks for the review!
I think these codes are trying to get all the nodes that participate in the computing, so BE num + CN num is correct in both modes.
Please give me a notice if I am wrong~

Reply:
It depends. In shared-nothing mode, compute nodes are not able to take the scanning tasks for the internal table. it will be inaccurate to take them into consideration in estimating the cost of broadcast v.s. shuffle. 

@satanson @ZiheLiu  what do you guys think?

Reply:
The codes are using the BE num to get 3 values:
1. Broadcast memory cost: rightOutput * beNum
2. Shuffle cpu cost: rightOutput / beNum(or dop)
3. CachePenaltyFactor while computing average probe cost in shuffle mode: mapsize / beNum (approximate formula)
I think 2 and 3 are not related to the scan task dop, so beNum + cnNum is correct. The first one depends on the if the source operator of left output is scan.

Reply:
Hello, I have added the run mode check while getting the memory cost of broadcast join. Could you please help me review this again?